editor.planx.uk/src/pages/Preview/Node.tsx

```diff
@@ -114,43 +112,10 @@ const Node: React.FC<any> = (props: Props) => {
         />
       );
     }
-    case TYPES.Confirmation: {
-      const payment: GovUKPayment | undefined =
-        passport.data?.[GOV_PAY_PASSPORT_KEY];
-
-      const details = {
-        "Planning application reference": payment?.reference ?? sessionId,
-
-        "Property address": passport.data?._address?.title,
-
-        "Application type": [
-          flowName.replace("Apply", "Application"),
-          getWorkStatus(passport),
-        ]
-          .filter(Boolean)
-          .join(" - "),
-
-        // XXX: If there is no payment we can't alternatively show Date.now() because it
-        //      will change after page refresh. BOPS submission time needs to be queryable.
-        Submitted: payment?.created_date
-          ? new Date(payment.created_date).toLocaleDateString("en-gb", {
-              day: "numeric",
-              month: "long",
-              year: "numeric",
-            })
-          : undefined,
-
-        "GOV.UK Payment reference": payment?.payment_id,
-      };
 
-      return (
-        <Confirmation
-          {...allProps}
-          details={objectWithoutNullishValues(details)}
-          color={{ text: "#000", background: "rgba(1, 99, 96, 0.1)" }}
-        />
-      );
-    }
```

Comment:
I personally think this was quite a strange pattern & unnecessarily extracted/pre-processed component content - and much prefer having a presentational component `Details` now that reads from the store.

The one trade-off is a less meanginful Confirmation Storybook because `details` can't be mocked as a prop any longer - but I think we can solve this with smarter state-mocking in future within the story itself?

Do say so if any objections to this approach or if you think the Storybook trade-off is too big !